whatshap/cli/stats.py

```diff
@@ -194,6 +198,46 @@ def add_heterozygous_variants(self, variants: int):
     def add_heterozygous_snvs(self, snvs: int):
         self.heterozygous_snvs += snvs
 
+    def get_nonoverlapping_blocks(self) -> List[PhasedBlock]:
+        """Split phase blocks into nonoverlapping subblocks"""
+        split_blocks = []
+        pos_sorted_blocks = sorted(
+            self.blocks, key=lambda b: (b.chromosome, b.leftmost_variant.position), reverse=True
+        )
+
+        # filter out blocks with only one variant
+        pos_sorted_blocks = [b for b in pos_sorted_blocks if len(b) > 1]
+
+        # iterate over blocks and split if overlapping until no blocks remain. non-overlapping
+        # blocks appended to split_blocks
+        while pos_sorted_blocks:
+            block = pos_sorted_blocks.pop()
+            if pos_sorted_blocks:
+                block_end = block.rightmost_variant.position
+                next_block = pos_sorted_blocks[-1]
+                next_block_start = next_block.leftmost_variant.position
+                next_block_end = next_block.rightmost_variant.position
+
+                # Check if next block overlapps current. If so split the current block.
+                if (block_end > next_block_start) and (block.chromosome == next_block.chromosome):
+                    block, new_block = block.split(next_block_start, next_block_end)
+
+                    # Update sorting if right-side block is added.
+                    if len(new_block) > 1:
+                        pos_sorted_blocks.append(new_block)
+                        pos_sorted_blocks = sorted(
```

Comment:
I was a bit worried because the entire list is resorted just because of one new element, but it is probably fine because Python’s sort is (as far as I know) fast with nearly sorted lists. (The alternative is to use a heap.)

I also think that one should only work on one chromosome at a time since blocks cannot span chromosomes, but I realize the original code did it like this, so let’s leave it as it is.